op-dispute-mon/mon/bonds/monitor.go

```diff
@@ -30,4 +40,80 @@ func (b *Bonds) CheckBonds(games []*types.EnrichedGameData) {
 	for addr, collateral := range data {
 		b.metrics.RecordBondCollateral(addr, collateral.Required, collateral.Actual)
 	}
+
+	for _, game := range games {
+		b.checkCredits(game)
+	}
+}
+
+func (b *Bonds) checkCredits(game *types.EnrichedGameData) {
+	addresses := make([]common.Address, len(game.Claims))
+	credits := make(map[common.Address]*big.Int)
+
+	// Check if the max duration has been reached for this game
+	duration := uint64(b.clock.Now().Unix()) - game.Timestamp
+	maxDurationReached := duration >= game.Duration
+
+	// Iterate over all *resolved* claims
+	claims := b.filterResolvedClaims(game)
```

Comment:
Seems like we should just inline the `if claim.Bond.Cmp(claimedBondFlag) != 0 {` condition into this for loop so we don't have to iterate through the claims array twice and build a partial copy.